controller/deployment.go

```diff
@@ -70,11 +70,17 @@ func (r *DeploymentRepo) Add(data interface{}) error {
 	if err != nil {
 		return err
 	}
-	// TODO: wrap all of this in a transaction once we move to pgx
-	if err := r.q.Enqueue(&que.Job{
-		Type: "deployment",
-		Args: args,
-	}); err != nil {
+
+	// Delete the deployment if we get ErrDeadConn because the client will
+	// retry on that error, but client retries will fail due to the
+	// deployment already existing in the database.
+	//
+	// TODO: wrap all of this in a transaction and drop the delete once we
+	//       move to pgx
+	job := &que.Job{Type: "deployment", Args: args}
+	if err := r.q.Enqueue(job); err == pgx.ErrDeadConn {
```

Comment:
Why just `ErrDeadConn`? If we're hackily emulating a transaction rollback, should any error trigger this?